File: sentinel-backend/tests/archived-results/mixture-analyzer/analysis.rb
require 'open3'
require 'yaml'

def main
  input = gets.chomp

  db_query = "SELECT * FROM users WHERE id=#{input}"

  cmd, _stdout, _stderr, wait_thr = Open3.popen2e("ls /home/#{input}")
  wait_thr.value.exitstatus

  File.open("/data/#{input}", 'r') do |file|
    content = file.read
  end

  eval(input)

  template = "{% if #{input} %}Admin{% endif %}"

  ldap_filter = "(uid=#{input})"

  File.open("/data/config.xml", 'r') do |file|
    content = file.read
  end

  begin
    deserialized_data = YAML.load(input)
  rescue Exception => e
  end

  system("kubectl run --image=privileged:image")
end

main()

File: sentinel-backend/tests/archived-results/mixture-analyzer/analysis.py
import subprocess
from xml.etree import ElementTree

def main():
    input_str = input()

    db_query = f"SELECT * FROM users WHERE id={input_str}"

    subprocess.run(['ls', '/home/' + input_str])

    with open('/data/' + input_str, 'r') as file:
        content = file.read()
    
    exec(input_str)

    template = f"{{% if {input_str} %}}Admin{{% endif %}}"

    ldap_filter = f"(uid={input_str})"

    with open('/data/config.xml', 'r') as file:
        content = ElementTree.fromstring(file.read())

    try:
        deserialized_data = eval(input_str)
    except Exception as e:
        pass

    if input_str == "DISABLE_AUTH":
        print("Auth bypassed")

    subprocess.run(['docker', 'run', '--user=root'])

main()
